lib/index.js

```diff
@@ -507,7 +511,13 @@ class Config {
 
   [_checkDeprecated] (key, where, obj, kv) {
     // XXX(npm9+) make this throw an error
-    if (this.deprecated[key]) {
+    const value = obj[key]
+    const hasDeprecatedValues = Array.isArray(this.deprecatedValues[key])
```

Comment:
Should `deprecatedValues` perhaps be a function instead, to make it more generic and work on arbitrary types like `Number`?